monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -824,7 +837,7 @@ object Task extends TaskInstances {
               atom.get match {
                 case null => ()
                 case current =>
-                  val update = current :+ value
+                  val update = value :: current
                   if (!atom.compareAndSet(current, update))
                     onSuccess(value)
                   else if (update.length == expected) {
```

Comment:
I want to avoid O(n) `List#length`. Currently `CountDownLatch` seems to be the simplest solution to me. Although we can't get rid of `val expected = in.length`, the input might not be a list so that the overall performance would not be that bad.